datumaro/components/extractor_tfds.py

```diff
@@ -138,12 +139,11 @@ def __call__(
         item: DatasetItem,
         state: namespace,
     ) -> None:
-        if self.filename_feature_name:
-            filename = tfds_example[self.filename_feature_name].numpy().decode("UTF-8")
+        data = tfds_example[self.feature_name].numpy()
+        if isinstance(data, np.ndarray):
+            item.media = Image.from_numpy(data=data)
         else:
-            filename = None
-
-        item.media = ByteImage(data=tfds_example[self.feature_name].numpy(), path=filename)
+            item.media = Image.from_bytes(data=data)
```

Comment:
The data has been already converted to `numpy` at line 142. Is there any case, where data is not `np.ndarray`?

Reply:
`numpy()` method tensorflow returns underlying data not just numpy.
It would return numpy if underlying data is numpy and it would return bytes if underlying data is bytes, and so on.